src/pages/docs/chat/api/javascript/room.mdx

```diff
@@ -0,0 +1,214 @@
+
+<Code>
+```javascript
+const room = await client.rooms.get('room:general');
+await room.attach();
+```
+</Code>
+
+---
+
+## Detach from a room <a id="detach"/>
+
+`room.detach()`
+
+Detaches from the room to stop receiving events.
+
+`detach(): Promise<void>`
+
+### Returns
+
+Returns a promise. On success, the promise is fulfilled and the room stops receiving events. On failure, the promise is rejected with an [`ErrorInfo`](/docs/chat/api/javascript/error-info).
+
+<Code>
+```javascript
+await room.detach();
+```
+</Code>
+
+---
+
+## Listen for RoomStatus changes <a id="onStatusChange"/>
+
+`room.onStatusChange()`
+
+Registers a listener that is called whenever the [`RoomStatus`](#roomstatus) changes.
+
+`onStatusChange(listener: RoomStatusListener): StatusSubscription`
+
+### Parameters
+
+Use the following parameters:
+
+| Parameter | Description | Type |
+| --------- | ----------- | ---- |
+| `listener` | Function called when the room status changes. | [`RoomStatusListener`](#roomstatuslistener) |
+
+#### RoomStatusListener
+
+A function that is called when the [`RoomStatus`](#roomstatus) of a room changes.
+
+`RoomStatusListener: (change: RoomStatusChange) => void`
+
+It uses the following parameters:
+
+| Parameter | Description | Type |
+| --------- | ----------- | ---- |
+| `change` | The change in `RoomStatus`. | [`RoomStatusChange`](#roomstatuschange) |
+
+#### RoomStatusChange
+
+An interface that represents a change in the [`RoomStatus`](#roomstatus) of a room.
+
+It has the following properties:
+
+| Property | Description | Type |
+| -------- | ----------- | ---- |
+| `previous` | The previous status of the room. | [`RoomStatus`](#roomstatus) |
+| `current` | The current status of the room. | [`RoomStatus`](#roomstatus) |
+| `error` | *Optional* The error that caused the change. | [`ErrorInfo`](/docs/chat/api/javascript/error-info) |
+
+### Returns
+
+Returns a `StatusSubscription` object with an `off()` method to remove the listener.
+
+<Code>
+```javascript
+room.onStatusChange((status) => {
+  console.log(`Room status changed to: ${status}`);
+});
+```
+</Code>
+
+#### StatusSubscription
+
+An interface that represents a subscription to status change events. It also provides a method of cleaning up subscriptions that are no longer needed.
+
+It has the following properties:
+
+| Property | Description | Type |
+| -------- | ----------- | ---- |
+| `off` | Unsubscribes from the status change events. It will ensure that no further status change events will be sent to the subscriber and that references to the subscriber are cleaned up. | Method |
+
+`off: () => void`
+
+---
+
+## Register a discontinuity handler <a id="onDiscontinuity"/>
+
+`room.onDiscontinuity()`
+
+Registers a handler that is called whenever a discontinuity is detected. Discontinuity occurs when a connection is interrupted and a `Room` cannot be resumed from its previous state.
```

Comment:
We should document what you should consider/do when tihs handler is called, e.g. calling `historyBeforeSubscribe` again, patching up history